doc/rst/language/spec/conversions.rst

```diff
@@ -218,54 +175,235 @@ type ``C`` is allowed when ``D`` is a subclass of ``C``.
-   is true otherwise.
+   The following code defines a function ``f`` accepting ``type t: int``
+   and then tries to pass ``int(8)`` to it. This will not compile,
+   because while an ``int(8)`` value can be implicitly converted to
+   ``int``, ``int(8)`` is not a subtype of ``int`` according to the above
+   definition.
 
--  An expression of a class type is taken to be false if it is nil and
-   is true otherwise.
+   .. code-block:: chapel
 
-.. _Implicit_Generic_Type_Conversions:
+      proc f(type t: int) { }
+      f(int(8));
+
+   .. BLOCK-test-chapeloutput
+
+      type-argument-conversion-error.chpl:2: error: unresolved call 'f(type int(8))'
+      type-argument-conversion-error.chpl:1: note: this candidate did not match: f(type t: int)
+      type-argument-conversion-error.chpl:2: note: because actual argument #1 with type 'int(8)'
+      type-argument-conversion-error.chpl:1: note: is passed to formal 't: int(64)'
+
+   *Example (type-argument-conversion.chpl)*
+
+   In contrast, this code demonstrates an implicit conversion that
+   does succeed because a child class is a subtype of a parent class, and
+   an ``owned`` class type is a subtype of an undecorated (generic
+   management) class type.
+
+   .. code-block:: chapel
+
+     class ParentClass { }
+     class ChildClass : ParentClass { }
+
+     proc g(type t: ParentClass) { }
+     g(owned ChildClass);
+
+
+.. _Implicit_Conversion_Init_Assign:
+
+Implicit Conversions for Initialization and Assignment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An implicit conversion for initialization or assignment occurs at each of
+the following program locations:
+
+-  In an assignment, the expression on the right-hand side of the
+   assignment is converted to the type of the expresssion on the
+   left-hand side of the assignment.
+
+-  In a variable or field declaration that is not a ref variable,
+   the initializing expression is converted to the type of the variable
+   or field. The initializing expression is the right-hand side of the
+   ``=`` in the declaration, if present, or in the field initialization
+   statement in an initializer.
+
+-  The return or yield expression within a function without a ``ref`` or
+   ``const ref`` return intent is converted to the return type of that
+   function.
+
+-  For a call to a function with a formal argument with ``out`` or
+   ``inout`` intent. The value of the formal argument is converted to the
+   type of the corresponding actual argument when setting that actual
+   with assignment or initialization (see :ref:`The_Out_Intent`).
+
+These implicit conversions can be implemented for record types by
+implementing ``init=`` and possibly ``=`` between two types as described in
+:ref:`Advanced_Copy_Initialization` and :ref:`Function_Overloading`.
+``init=`` will be called for initialization as
+described in :ref:`Split_Initialization` and other uses of ``=`` will
+invoke the ``=`` operator.
```

Comment:
This sentence slowed me down. How about:
```
init= will be called for .....
and the = operator will be invoked for the other uses of ``==``
```

Also, change `=` to `the operator =` a couple of lines earlier, so that I (the reader) can draw the parallel easily.